src/celeritas/neutron/interactor/detail/MomentumTransferSampler.hh

```diff
@@ -0,0 +1,383 @@
+#include "corecel/Types.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the momentum transfer (\f$ Q^{2} = -t \f$) of the neutron-nucleus
+ * elastic scattering.
+ *
+ * \note This performs the same sampling routine as in Geant4's
+ *  G4ChipsElasticModel and G4ChipsNeutronElasticXS where the differential
+ *  cross section for quark-exchange and the amplitude of the scattering are
+ *  parameterized in terms of the neutron momentum (GeV/c) in the lab frame.
+ */
+class MomentumTransferSampler
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Mass = units::MevMass;
+    using Momentum = units::MevMomentum;
+    using AtomicMassNumber = AtomicNumber;
+    //!@}
+
+  public:
+    // Construct with shared and target data, and the neutron momentum
+    inline CELER_FUNCTION
+    MomentumTransferSampler(NeutronElasticRef const& shared,
+                            IsotopeView const& target,
+                            Momentum neutron_p);
+
+    // Sample the momentum transfer
+    template<class Engine>
+    inline CELER_FUNCTION real_type operator()(Engine& rng);
+
+  private:
+    //// TYPES ////
+
+    using UniformRealDist = UniformRealDistribution<real_type>;
+
+    //// DATA ////
+
+    ChipsDiffXsCoefficients::ChipsArray const& par_;
+
+    // Mass of neutron and target
+    Mass neutron_mass_;
+    Mass target_mass_;
+
+    // Atomic mass number (A) of the target
+    AtomicMassNumber amass_;
+    bool heavy_target_{false};
+    // Momentum magnitude of the incident neutron (GeV/c)
+    Momentum neutron_p_;
+    // Maximum momentum transfer for the elastic scattering
+    real_type max_q_sq_;
+    // Parameters for the CHIPS differential cross section
+    ExchangeParameters par_q_sq_;
+
+    //// HELPER FUNCTIONS ////
+
+    // Sample the slope
+    template<class Engine>
+    inline CELER_FUNCTION real_type sample_q_sq(real_type radius, Engine& rng)
+    {
+        return -std::log(1 - radius * generate_canonical(rng));
+    }
+
+    // Calculate the maximum momentum transfer
+    inline CELER_FUNCTION real_type calc_max_q_sq(Momentum p) const;
+
+    // Calculate parameters used in the quark-exchange process
+    inline CELER_FUNCTION ExchangeParameters calc_par_q_sq(Momentum p) const;
+
+    //// COMMON PROPERTIES ////
+
+    // Covert from clhep::MeV value to clhep::GeV value
+    static CELER_CONSTEXPR_FUNCTION real_type to_gev() { return 1e-3; }
+
+    // S-wave limit for neutron, log(p) < -4.3 (GeV/c) (kinetic energy < 0.1
+    // MeV)
+    static CELER_CONSTEXPR_FUNCTION Momentum s_wave_limit()
+    {
+        return Momentum{13.568559};
```

Comment:
If you are going add 6 digits, might as well add them all, right?

```suggestion
        return Momentum{13.568559012200934}; // = exp(-4.3) * 1000
```

Reply:
Updated as suggested.  Thanks!